tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/CalDavEventAttendanceRepository.java

```diff
@@ -0,0 +1,272 @@
+                .switchIfEmpty(Mono.error(() -> new RuntimeException("Unable to find user in Dav server with username '%s'".formatted(username.asString()))));
+        } catch (AddressException e) {
+            LOGGER.debug("Failed to get user DAV information", e);
+            return Mono.empty();
+        }
+    }
+
+    @Override
+    public Publisher<CalendarEventReplyResults> setAttendanceStatus(Username username,
+                                                                    AttendanceStatus attendanceStatus,
+                                                                    BlobIds eventBlobIds,
+                                                                    Optional<LanguageLocation> maybePreferredLanguage) {
+        MailboxSession session = sessionProvider.createSystemSession(username);
+
+        return toDavUser(username).flatMapMany(davUser -> blobIdsAsFlux(eventBlobIds)
+                    .flatMap(blobId -> updateAttendanceStatusForEventBlob(blobIdFromRefinedId(blobId), attendanceStatus, davUser, session)));
+    }
+
+    private Mono<CalendarEventReplyResults> updateAttendanceStatusForEventBlob(BlobId blobId, AttendanceStatus attendanceStatus, DavUser davUser, MailboxSession session) {
+        PartStat partStat = attendanceStatus.toPartStat().orElseThrow();
+        return fetchCalendarObject(blobId, davUser, session)
+            .flatMap(targetCalendarObject -> updateCalendarObject(davUser, targetCalendarObject.uri(), partStat))
+            .thenReturn(CalendarEventReplyResults$.MODULE$.done(blobId))
+            .onErrorResume(e -> Mono.just(CalendarEventReplyResults$.MODULE$.notDone(blobId, e, session)));
+    }
+
+    private Mono<Void> updateCalendarObject(DavUser davUser, URI calendarObjectURI, PartStat partStat) {
+        return davClient.updateCalendarObject(davUser, calendarObjectURI, calendarObject -> createUpdatedCalendar(calendarObject, davUser.username(), partStat));
+    }
+
+    private Mono<DavCalendarObject> doGetCalendarObjectContainingVEvent(DavUser davUser, String eventUid) {
+        return davClient.getCalendarObjectContainingVEvent(davUser, eventUid)
+            .switchIfEmpty(Mono.error(() -> new RuntimeException("Unable to find any calendar objects containing VEVENT with id '%s'".formatted(eventUid))));
+    }
+
+    private DavCalendarObject createUpdatedCalendar(DavCalendarObject calendarObject, String targetAttendeeEmail, PartStat partStat) {
+        Calendar updatedCalendarData = calendarObject.calendarData().copy();
+        LOGGER.trace("Calendar to update: {}", calendarObject.calendarData());
+
+        List<VEvent> updatedVEvents = updatedCalendarData.<VEvent>getComponents(Component.VEVENT)
+            .stream().map(vEvent ->
+                vEvent.getAttendees()
+                    .stream()
+                    .filter(attendee -> targetAttendeeEmail.equals(attendee.getCalAddress().toASCIIString().substring(MAILTO_PREFIX_LENGTH)))
```

Comment:
IMO we should substring if, and only if it starts with `mailto:`

I'd sujest

```suggestion
                    .filter(attendee -> attendee.getCalAddress().toASCIIString().equalsIgnoreCase("mailto:" + targetAttendeeEmail))
```